public/functions.php

```diff
@@ -0,0 +1,116 @@
+			return str_replace( 'public/', '', trim( $yaml['path'], '/' ) );
+		}
+	}
+
+	return 'trunk-git/build';
+}
+
+/**
+ * Detects and sets the location of the WordPress core and content directories to use.
+ *
+ * The function sets the 'ABSPATH' constant based on the function parameter or the
+ * {@see _wordpressdev_detect_core_path_relative()} function.
+ *
+ * The function also sets the 'WP_CONTENT_DIR' constant, based on a hard-coded content path
+ * 'content'.
+ *
+ * @access private
+ *
+ * @param string $core_path_relative    Optional. WordPress core path, relative to the 'public' directory. Default
+ *                                      is determined by {@see _wordpressdev_detect_core_path_relative()}.
+ * @param string $content_path_relative Optional. WordPress content path, relative to the 'public' directory. Default
+ *                                      is 'content'.
+ */
+function _wordpressdev_set_directory_constants( $core_path_relative = '', $content_path_relative = 'content' ) {
+	if ( empty( $core_path_relative ) ) {
+		$core_path_relative = _wordpressdev_detect_core_path_relative();
+	}
+
+	// Define custom 'WP_CORE_PATH_RELATIVE' constant, for WordPress core directory relative to 'public'.
+	define( 'WP_CORE_PATH_RELATIVE', $core_path_relative );
+
+	// Define custom 'WP_CONTENT_PATH_RELATIVE' constant, for WordPress content directory relative to 'public'.
+	define( 'WP_CONTENT_PATH_RELATIVE', $content_path_relative );
+
+	// Define 'ABSPATH'.
+	if ( ! defined( 'ABSPATH' ) ) {
+		define( 'ABSPATH', dirname( __FILE__ ) . '/' . WP_CORE_PATH_RELATIVE . '/' );
+	}
+
+	// Define 'WP_CONTENT_DIR'.
+	if ( ! defined( 'WP_CONTENT_DIR' ) ) {
+		define( 'WP_CONTENT_DIR', dirname( __FILE__ ) . '/' . WP_CONTENT_PATH_RELATIVE );
+	}
+}
+
+/**
+ * Defines additional URL constants for WordPress core and content directories to use.
+ *
+ * The function defines these constants based on the 'WP_HOME' constant. It should be called whenever that
+ * constant has been set, which typically is in either 'wp-config.php' or, in multisite, from the 'sunrise.php'
+ * drop-in or via the 'ms_loaded' action.
+ *
+ * The function will try to set 'WP_HOME' initially if it is not available yet. This may fail when accessing
+ * via WP-CLI without the --url argument.
+ *
+ * @access private
+ */
+function _wordpressdev_set_url_constants() {
+	// Define 'WP_HOME' if possible.
+	if ( ! defined( 'WP_HOME' ) ) {
+		if ( defined( 'MULTISITE' ) && MULTISITE && ! empty( $GLOBALS['current_blog'] ) ) {
+			// In multisite, rely on the $current_blog global if yet available.
+			$domain = $GLOBALS['current_blog']->domain;
+			$path   = ! empty( $GLOBALS['current_blog']->path ) && '/' !== $GLOBALS['current_blog']->path ? $GLOBALS['current_blog']->path : '';
+			define( 'WP_HOME', 'https://' . $domain . $path );
+		} else {
+			// Otherwise, rely on the 'HTTP_HOST' server variable.
+			if ( ! isset( $_SERVER['HTTP_HOST'] ) ) {
+				return;
```

Comment:
This `return` will short-circuit the remaining constants defined below. Sure they depend on `WP_HOME` but seems a bit prone to a future bug if other constants are added. I suggest moving the `isset( $_SERVER['HTTP_HOST']` into an `elseif` to replace the `else`, and then below for the two `if` statements check if `WP_HOME` is defined.

Reply:
I intended this function to set only those constants that are for URLs, and thus depend on the home URL being known. However your point definitely makes sense regardless. I'll take out the part where it sets up `WP_HOME` conditionally, and put it into a separate function. The existing function can then receive the home URL to use as (optional) parameter, similar like the function to set directory constants.